File: pull.php
<?php

require 'vendor/autoload.php';

use GuzzleHttp\Client;
use GuzzleHttp\Exception\RequestException;
use GuzzleHttp\Psr7\Response;

$API_BASE_URL = "http://YOURIP:5700";
$API_TOKEN = "YOUR_Authorization";

// 设置日志
$log = new Monolog\Logger('logger');
$log->pushHandler(new Monolog\Handler\StreamHandler('php://stdout', Monolog\Logger::INFO));

function main() {
    global $log;
    try {
        $record_data = get_data("/api/scripts?t=");
        $log->info("获取到的body code: " . $record_data["code"]);

        foreach ($record_data["data"] as $val) {
            try {
                if (isset($val["children"])) {
                    foreach ($val["children"] as $child_val) {
                        $child_data = get_data("/api/scripts/" . $child_val["title"] . "?path=" . $child_val["parent"] . "&t=");
                        download_file($child_data["data"], $child_val["parent"], $child_val["title"]);
                    }
                } else {
                    $child_data = get_data("/api/scripts/" . $val["title"] . "?path=&t=");
                    download_file($child_data["data"], "", $val["title"]);
                }
            } catch (Exception $e) {
                $log->error("处理子项目时出错: " . $e->getMessage());
            }
        }
        $log->info("执行完毕");
    } catch (Exception $e) {
        $log->error("An error occurred: " . $e->getMessage());
    }
}

function get_data($url_str) {
    global $API_BASE_URL, $API_TOKEN, $log;
    try {
        $timestamp = strval(intval(microtime(true) * 1000));
        $full_url = $API_BASE_URL . $url_str . $timestamp;
        $log->info($full_url);
        $headers = ["Authorization" => "Bearer " . $API_TOKEN];
        $client = new Client();
        $response = $client->get($full_url, ["headers" => $headers]);
        $statusCode = $response->getStatusCode();
        if ($statusCode >= 200 && $statusCode < 300) {
            return json_decode($response->getBody(), true);
        } else {
            throw new Exception("Request error: " . $response->getReasonPhrase());
        }
    } catch (RequestException $e) {
        throw new Exception("Request error: " . $e->getMessage());
    }
}

function download_file($body, $path, $name) {
    global $log;
    try {
        $file_path = "./" . $name;
        if ($path) {
            if (!file_exists($path)) {
                mkdir($path, 0777, true);
                chmod($path, 0777);
            }
            $file_path = $path . "/" . $name;
        }
        file_put_contents($file_path, $body);
    } catch (Exception $e) {
        throw new Exception("File download error: " . $e->getMessage());
    }
}

main();

?>

File: pull.java
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.logging.Level;
import java.util.logging.Logger;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class Main {
    private static final String API_BASE_URL = "http://YOURIP:5700";
    private static final String API_TOKEN = "YOUR_Authorization";
    private static final Logger logger = Logger.getLogger(Main.class.getName());

    public static void main(String[] args) {
        try {
            String recordData = getData("/api/scripts?t=");
            logger.info("获取到的body code: " + recordData.get("code"));

            for (Object val : recordData.get("data")) {
                try {
                    if (((Map<String, Object>) val).containsKey("children")) {
                        for (Object childVal : ((Map<String, Object>) val).get("children")) {
                            Map<String, Object> childData = getData("/api/scripts/" + ((Map<String, Object>) childVal).get("title") + "?path=" + ((Map<String, Object>) childVal).get("parent") + "&t=");
                            downloadFile((String) childData.get("data"), (String) ((Map<String, Object>) childVal).get("parent"), (String) ((Map<String, Object>) childVal).get("title"));
                        }
                    } else {
                        Map<String, Object> childData = getData("/api/scripts/" + ((Map<String, Object>) val).get("title") + "?path=&t=");
                        downloadFile((String) childData.get("data"), "", (String) ((Map<String, Object>) val).get("title"));
                    }
                } catch (Exception e) {
                    logger.log(Level.SEVERE, "处理子项目时出错: " + e.getMessage(), e);
                }
            }
            logger.info("执行完毕");
        } catch (Exception e) {
            logger.log(Level.SEVERE, "An error occurred: " + e.getMessage(), e);
        }
    }

    private static Map<String, Object> getData(String urlStr) throws IOException {
        try {
            String timestamp = String.valueOf(System.currentTimeMillis());
            String fullUrl = API_BASE_URL + urlStr + timestamp;
            logger.info(fullUrl);
            OkHttpClient client = new OkHttpClient();
            Request request = new Request.Builder()
                    .url(fullUrl)
                    .addHeader("Authorization", "Bearer " + API_TOKEN)
                    .build();
            try (Response response = client.newCall(request).execute()) {
                if (!response.isSuccessful()) {
                    throw new IOException("Unexpected response code: " + response);
                }
                String responseBody = response.body().string();
                return new ObjectMapper().readValue(responseBody, new TypeReference<Map<String, Object>>() {});
            }
        } catch (IOException e) {
            throw new IOException("Request error: " + e.getMessage(), e);
        }
    }

    private static void downloadFile(String body, String path, String name) throws IOException {
        try {
            String filePath = Paths.get(".", name).toString();
            if (!path.isEmpty()) {
                Path directoryPath = Paths.get(path);
                if (!Files.exists(directoryPath)) {
                    Files.createDirectories(directoryPath);
                    Files.setPosixFilePermissions(directoryPath, PosixFilePermissions.fromString("rwxrwxrwx"));
                }
                filePath = Paths.get(path, name).toString();
            }
            try (FileOutputStream fos = new FileOutputStream(filePath)) {
                fos.write(body.getBytes(StandardCharsets.UTF_8));
            }
        } catch (IOException e) {
            throw new IOException("File download error: " + e.getMessage(), e);
        }
    }
}